devtools/client/framework/toolbox-init.js

```diff
@@ -51,22 +62,32 @@ if (url.search.length > 1) {
         DebuggerServer.addBrowserActors();
       }
       let client = new DebuggerClient(DebuggerServer.connectPipe());
-      Task.spawn(function*() {
+      toolboxOpened = Task.spawn(function*() {
         yield client.connect();
         // Creates a target for a given browser iframe.
         let response = yield client.getTab({ tab });
         let form = response.tab;
         let target = yield TargetFactory.forRemoteTab({client, form, chrome: false});
         let options = { customIframe: host };
-        yield gDevTools.showToolbox(target, tool, Toolbox.HostType.CUSTOM, options);
+        return gDevTools.showToolbox(target, tool, Toolbox.HostType.CUSTOM, options);
       });
     }
   } else {
-    targetFromURL(url).then(target => {
+    toolboxOpened = Task.spawn(function*() {
+      let target = yield targetFromURL(url);
       let options = { customIframe: host };
       return gDevTools.showToolbox(target, tool, Toolbox.HostType.CUSTOM, options);
-    }).then(null, e => {
-      window.alert("Unable to start the toolbox:" + e.message);
+    }).catch(e => {
+      window.alert("Unable to start the toolbox: " + e.message);
     });
```

Comment:
This *catch* call should rethrow the exception, otherwise this *toolboxOpened* promise will be resolved instead of rejected. Alternately, move the *catch* call to the `toolboxOpened.then()` chain below, so it catches and reports exceptions in both *toolboxOpened* promises.